### PR TITLE
Enable CUDA builds using conda-forge packages

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### Linux 64
None so far.

### Windows
- [x] Provide drivers for CI -- see https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/129
- [x] `nvcc` does not exist yet -- see https://github.com/conda-forge/nvcc-feedstock/pull/49
- [x] Update matrix for `cuda_compiler` and `cuda_version` in `conda-forge-pinning`  -- see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/914
  - Added manual migrator with commit fc40e07

Note: We might need to drop CUDA 9.2 on Windows because it requires `vs2015`, now deprecated in conda-forge.

### MacOS
None so far. Cuda not available.

### Linux PPC
- [ ] `nvcc` does not exist yet
- [ ] Update matrix for `cuda_compiler` and `cuda_version` in `conda-forge-pinning` 

### ARM
None so far. Cuda not available.

### CODE REVIEW

File: .ci_support/linux_64_cuda_compiler_version10.0numpy1.16python3.6.____cpython.yaml [hunk 1]
```diff
@@ -35,8 +35,7 @@ target_platform:
 zip_keys:
 - - c_compiler_version
   - cxx_compiler_version
-- - cdt_name
```
Comment (jakirkham) [inline .ci_support/linux_64_cuda_compiler_version10.0numpy1.16python3.6.____cpython.yaml:38]: May want to manually add this back in. Also is missing in the other variant files

**Comment by jaimergp** [comment]
I am going to merge later today so we can have something to test locally. I will change the label so it's `test` to mimic `cudatoolkit`. This might need an update in some of our downstream CI, @jchodera, but I will handle that too.

This is being labeled as 7.5.0, btw, but shouldn't cause too much trouble given that we are in a different subchannel anyway.

PowerPC will be addressed in a separate PR since it's not as urgent.

**Comment by jakirkham** [comment]
Thanks so much for working on this Jaime! 😄

On `ppc64le` we are still missing some `cudatoolkit` packages ( https://github.com/conda-forge/cudatoolkit-feedstock/issues/12 ). So we would probably want to resolve that first (if we are interested in those versions).

**Comment by jaimergp** [comment]
As far as OpenMM is concerned we only need _one_ CUDA build that works nicely at Summit. We are hoping 10.2 is that version 😬 

If it's not, I'll put some work on 10.1 and 10.0 (and I guess 9.2 for free?). Before that, I will submit PRs for `nvcc` and `conda-forge-pinning` though (even if it's only for 10.2).

Btw, while I have your attention, can you take a look at https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/131 ? Ideally there's a Nvidia-approved way to get `nvcuda.dll` installed on GPU-less machines :)

**Comment by jchodera** [comment]
@jaimergp Did we ever get the `ppc64le` issues sorted out? Is it just the missing `cudatoolkit` package we're waiting on?

**Comment by jaimergp** [comment]
Docker images and cudatoolkit packages are in place. We only need the `ppc64le` nvcc wrapper (and I guess the relevant updates in the pinnings). However, I wanted to wait a bit to do this until we could run the full CI suite with the conda compilers (GCC9), which we are doing now in https://github.com/openmm/openmm/pull/2964. There are some non-trivial errors that'd need to be fixed before releasing the ppc64le builds.

**Comment by jchodera** [comment]
@jaimergp Would you be able to comment on what non-trivial errors remain?

**Comment by jaimergp** [comment]
CI for `master` passes for a configuration compatible with conda-forge, so it _should_ work for the upcoming 7.5.1, discussed here https://github.com/openmm/openmm/issues/2993

**Pull request merged**